pgrx/src/memcx.rs

```diff
@@ -45,3 +62,54 @@ where
 
     f(&memcx)
 }
+
+#[cfg(all(feature = "nightly", feature = "pg16"))]
+unsafe impl<'mcx> std::alloc::Allocator for &MemCx<'mcx> {
```

Comment:
I think it would be slightly nicer to do
```suggestion
#[cfg(all(feature = "nightly", feature = "pg16"))]
mod nightly {
    use std::slice;
    use super::*;

    unsafe impl<'mcx> std::alloc::Allocator for &MemCx<'mcx> {
```

Reply:
This would also make it easier to add other std imports for just this segment of code.